File: app/models/redeem_reward_record.rb
require 'util/constant'

class RedeemRewardRecord
  include DataMapper::Resource

  Modes = [:reward, :prize]
  
  property :id, Serial
  property :reward_id, Integer, :required => true, :default => 0
  property :venue_id, Integer, :required => true, :default => 0
  property :points, Integer, :required => true, :default => 0
  property :mode, Enum[:reward, :prize], :required => true, :default => :reward
  property :created_ts, DateTime, :required => true, :default => ::Constant::MIN_TIME
  property :update_ts, DateTime, :default => ::Constant::MIN_TIME
  property :deleted_ts, ParanoidDateTime
  #property :deleted, ParanoidBoolean, :default => false
  
  attr_accessible :reward_id, :venue_id, :points, :mode, :created_ts, :update_ts
  
  belongs_to :merchant
  belongs_to :customer
  belongs_to :user
end

File: public/javascripts/mobile/app/model/RedeemRewardRecord.js
Ext.require(['Genesis.model.Merchant', 'Genesis.model.User'], function()
{
   Ext.define('Genesis.model.RedeemRewardRecord', {
      extend : 'Ext.data.Model',
      id : 'RedeemRewardRecord',
      fields : ['reward_id', 'points', 'time'],
      belongsTo : ['Genesis.model.Merchant', 'Genesis.model.User']
   });
});
